connector-schemas/kafka/table.json

```diff
@@ -22,18 +22,38 @@
                                 "earliest",
                                 "latest"
                             ]
+                        },
+                        "read_mode": {
+                            "type": "string",
+                            "description": "Type of reading behavior for Kafka Source",
+                            "enum": [
+                                "read_committed",
+                                "read_uncommitted"
+                            ]
                         }
                     },
                     "required": [
-                        "offset"
+                        "offset",
+                        "read_mode"
                     ],
                     "additionalProperties": false
                 },
                 {
                     "type": "object",
                     "title": "Sink",
                     "properties": {
+                        "commit_mode": {
+                            "type": "string",
+                            "description": "Type of committing behavior for Kafka Sink",
```

Comment:
I think this description could be expanded a bit to say that this enables transactional semantics and requires that transactional support be enabled in Kafka